### DISCUSSION TITLE
Sending the View itself as a Behavior's CommandParameter

### DISCUSSION THREAD

**Discussion by Cfun1** [xamarin/XamarinCommunityToolkit#1040] [Q&A] [answered]
It seems not possible to bind the object itself implementing the behavior as a `CommandParameter`, tried binding with `FindAncestor` and `Self` and both didn't work, I see the only option in this case is to set a `x:Name and Source="{x:Reference name}"` which I prefer not to use.

###  Reference
https://github.com/xamarin/Xamarin.Forms/issues/8584

### When this could be needed ?
When you have some attached properties to your View and you want to access them in the behavior `Command` logic.

~jBijsterboschNL  any ideas other then sending it as reference? (I first came across this scenario when was using `UserStoppedTypingBehavior` :) )~

**Comment by pictos** [comment, marked as answer]
@Cfun1 if we think about MVVM, we shouldn't pass the View as CommandParameter, because the VM shouldn't have anything of the View on it.

**Reply by Cfun1** [reply]
Good point, then I should think about a redesign.
